src/main/java/org/svydovets/dao/GenericJdbcDAO.java

```diff
@@ -187,18 +198,118 @@ private PreparedStatement prepareSelectStatement(EntityKey<?> entityKey, Connect
         }
     }
 
-    private <T> T createEntityFromResultSet(EntityKey<T> entityKey, ResultSet resultSet) {
+    private <T> T createEntityFromResultSet(Class<T> clazz, ResultSet resultSet) {
         try {
-            T entity = entityKey.entityType().getConstructor().newInstance();
-            ResultSetParser.parseForEntity(entity, resultSet);
+            T entity = clazz.getConstructor().newInstance();
+            try {
+                for (Field field : entity.getClass().getDeclaredFields()) {
+                    field.setAccessible(true);
+                    if (isEntityField(field)) {
+                        var joinClazz = field.getType();
+                        var joinColumnName = ParameterNameResolver.resolveColumnName(field);
+                        var joinColumnValue = resultSet.getObject(joinColumnName);
+                        var entityKey = new EntityKey<>(joinClazz, joinColumnValue);
+                        var joinEntity = loadFromDB(entityKey);
+                        field.set(entity, joinEntity);
+                    } else if (isEntityCollectionField(field)) {
+                        var joinClazz = getJoinCollectionEntityType(field);
+                        var entityFieldInJoinClazz = getJoinClazzField(clazz, joinClazz);
+                        var joinEntityId = resultSet.getObject(ParameterNameResolver.getIdFieldName(joinClazz));

```

Comment:
Lets take as example "Person <-> Note" relationship

In this case? here we want to get an id of person, to perform a "select * from notes where person_id = ?", right? But you pass the name of id field for "joinClazz" which is "Note", isn't?